static/app/views/issueDetails/issueDetailsTour.tsx

```diff
@@ -0,0 +1,46 @@
+import {createContext, useContext} from 'react';
+
+import {TourElement, type TourElementProps} from 'sentry/components/tours/components';
+import type {TourContextType} from 'sentry/components/tours/tourContext';
+
+export const enum IssueDetailsTour {
+  /** Onboarding for trends and aggregates, the graph, and tag distributions */
+  AGGREGATES = 'issue-details-aggregates',
+  /** Onboarding for date/time/environment filters */
+  FILTERS = 'issue-details-filters',
+  /** Onboarding for event details, event navigation, main page content */
+  EVENT_DETAILS = 'issue-details-event-details',
+  /** Onboarding for event navigation; next/previous, first/last/recommended events */
+  NAVIGATION = 'issue-details-navigation',
+  /** Onboarding for workflow actions; resolution, archival, assignment, priority, etc. */
+  WORKFLOWS = 'issue-details-workflows',
+  /** Onboarding for activity log, issue tracking, solutions hub area */
+  SIDEBAR = 'issue-details-sidebar',
+}
+
+export const ORDERED_ISSUE_DETAILS_TOUR = [
+  IssueDetailsTour.AGGREGATES,
+  IssueDetailsTour.FILTERS,
+  IssueDetailsTour.EVENT_DETAILS,
+  IssueDetailsTour.NAVIGATION,
+  IssueDetailsTour.WORKFLOWS,
+  IssueDetailsTour.SIDEBAR,
+];
+
+export const IssueDetailsTourContext = createContext<TourContextType<IssueDetailsTour>>({
+  currentStepId: null,
+  isAvailable: false,
+  isRegistered: false,
+  orderedStepIds: ORDERED_ISSUE_DETAILS_TOUR,
+  dispatch: () => {},
+  handleStepRegistration: () => () => {},
+});
```

Comment:
Default initializers are not a good idea here, the reason being that they can hide bugs where someone forgets to wrap their tour in an actual provider, and then calls to dispatch or handleRegistration will simply noop. An alternative is to define this as `createContext<T | null>(null)` and handle that assertion inside your `useIssueDetailsTour` where you can assert a non null context like

```tsx
export function useIssueDetailsTour(): TourContextType<IssueDetailsTour> {
  const context = useContext(IssueDetailsTourContext);

  if(context === null) throw 'oops, you forgot to use a context'

  return context;
}
```

Reply:
That is a really good call, removing this initializer 👍